### PR TITLE
Make it easy to run tests with address sanitizer etc

### PR DESCRIPTION
**Pull request by kjellwinblad**
This PR contains the following changes:

* A script that makes it easy to run the tests with address sanitizer in Docker
* A fix for a memory leak that was found by address sanitizer
* Better descriptions for how to run the tests and some cleanups in the README.md file

### CODE REVIEW

**Review by starbelly — APPROVED** [review]
Worked like a charm!

File: c_src/sasl_auth.c [hunk 1]
```diff
@@ -340,8 +340,10 @@ static ERL_NIF_TERM sasl_kinit(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF_T
 
     krb5_free_cred_contents(context, &creds);
     krb5_free_principal(context, principal);
+    if (options != NULL) {
```
Comment (starbelly) [inline c_src/sasl_auth.c:343]: Thank you address sanitizer!

**Review by starbelly — CHANGES_REQUESTED** [review]
I've removed my approval, I'm having trouble running this now. I'll have to look in a bit as to what the issue is.

**Comment by kjellwinblad** [comment]
@starbelly thanks for having a look. Maybe I can help you to find the problem if you share what type of error you got?

Btw, I added a commit (94e2375) that makes it easy to run the tests in different docker images. Is the `scripts/docker_build.sh` and `scripts/docker_run.sh` scripts still useful for you after this commit?

**Comment by starbelly** [comment]
> @starbelly thanks for having a look. Maybe I can help you to find the problem if you share what type of error you got?
> 
> Btw, I added a commit ([94e2375](https://github.com/kafka4beam/sasl_auth/commit/94e23755bedab58f4599e76b9ef200d0ac3fb16d)) that makes it easy to run the tests in different docker images. Is the `scripts/docker_build.sh` and `scripts/docker_run.sh` scripts still useful for you after this commit?


I see the problem this morning. There's some places where we have `x86_64-pc-linux-gnu` hard coded, and on my m1 it's `x86_64-pc-linux-gnu`. 

That said let me test from an x86 machine again... 

Edit: 

Yeah, fine on x86, with multiple runs.

**Comment by kjellwinblad** [comment]
@starbelly 
> I see the problem this morning. There's some places where we have x86_64-pc-linux-gnu hard coded, and on my m1 it's x86_64-pc-linux-gnu.

I have removed these strings ([9dc4795](https://github.com/kafka4beam/sasl_auth/pull/17/commits/9dc47958dd15a032d4516a23d600fe4ca06fb3ed)). Does it work better now?

Edit: Did you hit the problem when you ran `./scripts/address_sanitizer/setup.sh && ./scripts/address_sanitizer/run.sh` directly in Mac OS or when you ran `./scripts/setup_and_run_rebar3_ct_in_docker.sh asan`?

**Comment by starbelly** [comment]
> Did you hit the problem when you ran `./scripts/address_sanitizer/setup.sh && ./scripts/address_sanitizer/run.sh` directly in Mac OS or when you ran `./scripts/setup_and_run_rebar3_ct_in_docker.sh asan`?

Running the docker script... trying again now...

**Review by starbelly — APPROVED** [review]
Works solid on both x86 and aarch64. This helps a lot!

**Comment by starbelly** [comment]
@kjellwinblad Approved, but left it open so you can squash your changes if you like.

**Pull request merged**